### ISSUE TITLE
Catboost fails on GPU

### ISSUE THREAD

**Issue by gilgils** [catboost/catboost#1796]
Problem: I am trying to run CatboostClassifier on GPU and it fails. The "fit" seems to run, and all the iterations are done, but probably something it the ends of the process causing it to collapse, reporting "Process finished with exit code -1073740791 (0xC0000409)". 
catboost version: 0.26
Operating System:
CPU: AMD Ryzen 7 5800HS
GPU: RTX3060 for Notebooks.

I do not think a code will help, but here it is:

import numpy as np
from catboost import CatBoostClassifier
from catboost import Pool
import os

os.environ["CUDA_VISIBLE_DEVICES"] = "0"
A = np.random.randn(1000, 10)
y = (np.random.randn(1000, 1) < 0).astype('int')

data = Pool(data=A, label=y)
model = CatBoostClassifier(iterations=20, task_type='GPU')

print('fit')
model.fit(data)
print('done')

output:

fit
Learning rate set to 0.5
0:	learn: 0.6707387	total: 3.82ms	remaining: 72.5ms
1:	learn: 0.6569982	total: 6.78ms	remaining: 61.1ms
2:	learn: 0.6469501	total: 9.41ms	remaining: 53.4ms
3:	learn: 0.6354006	total: 12ms	remaining: 47.9ms
4:	learn: 0.6237046	total: 14.7ms	remaining: 44ms
5:	learn: 0.6124821	total: 17.3ms	remaining: 40.4ms
6:	learn: 0.6048968	total: 19.9ms	remaining: 36.9ms
7:	learn: 0.5969783	total: 22.6ms	remaining: 33.9ms
8:	learn: 0.5897941	total: 25.3ms	remaining: 30.9ms
9:	learn: 0.5794357	total: 27.9ms	remaining: 27.9ms
10:	learn: 0.5715412	total: 30.5ms	remaining: 25ms
11:	learn: 0.5643271	total: 33.2ms	remaining: 22.1ms
12:	learn: 0.5533641	total: 35.7ms	remaining: 19.2ms
13:	learn: 0.5449307	total: 38.1ms	remaining: 16.3ms
14:	learn: 0.5321642	total: 40.5ms	remaining: 13.5ms
15:	learn: 0.5248699	total: 43.1ms	remaining: 10.8ms
16:	learn: 0.5151945	total: 45.6ms	remaining: 8.05ms
17:	learn: 0.5089163	total: 48.3ms	remaining: 5.37ms
18:	learn: 0.5022792	total: 50.8ms	remaining: 2.67ms
19:	learn: 0.4956213	total: 53.4ms	remaining: 0us

Process finished with exit code -1073740791 (0xC0000409)

You can see that it doesn't reach to the "print('done')"line.

**Comment by DanielLumb** [comment]
This is duplicate of #1735 (and a few other related issues), already fixed and waiting to be released in 26.1.

**Comment by kizill** [comment]
Published 0.26.1 with fix.
